src/descriptor/mod.rs

```diff
@@ -393,29 +393,28 @@ impl<Pk: MiniscriptKey + ToPublicKey> Descriptor<Pk> {
         }
     }
 
-    /// Get the `scriptCode` as [defined by
-    /// bip-0143](https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#specification),
-    /// used to generate sighashes for spending Segwit outputs.
+    /// Get the `scriptCode` of a transaction output.
     ///
-    /// # Errors
-    /// - If the script descriptor type is non-segwit, as the script code is only defined for
-    /// segregated witness transaction outputs.
-    pub fn script_code(&self) -> Result<Script, Error> {
+    /// The `scriptCode` is the Script of the previous transaction output being serialized in the
+    /// sighash when evaluating a `CHECKSIG` & co. OP code.
+    pub fn script_code(&self) -> Script {
         match *self {
-            // Spending non-segwit outputs requires "legacy" sighash and don't use bip143's `scriptCode`.
+            // For "legacy" outputs, it is defined as the txo's scriptPubKey.
```

Comment:
For `Sh` it should be the redeem script. Otherwise I think this is ok

Reply:
Thanks, amended.